### PR TITLE
CI: Fix CMake definition of streaming service options

### PR DESCRIPTION
**Pull request by PatTheMav**
### Description
On master branch, client information for Twitch, YouTube and Restream integration are set as CMake cache variables. The hashes can contain otherwise "illegal" characters such as backticks which need to be enclosed in a string, but also cannot be subject to variable substition.

Using single quotes should fix this.

### Motivation and Context
Fix CI workflow runs on master branch.

### How Has This Been Tested?
Alas as this is reliant on OBS' client information, this has to be merged and tested on the master branch.

### Types of changes
- Bug fix (non-breaking change which fixes an issue)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: CI/linux/02_build_obs.sh [hunk 1]
```diff
@@ -34,15 +34,15 @@ _configure_obs() {
     check_ccache
 
     if [ "${TWITCH_CLIENTID}" -a "${TWICH_HASH}" ]; then
-        TWITCH_OPTIONS="-DTWITCH_CLIENTID=\"${TWITCH_CLIENTID}\" -DTWITCH_HASH=\"${TWITCH_HASH}\""
+        TWITCH_OPTIONS="-DTWITCH_CLIENTID='${TWITCH_CLIENTID}' -DTWITCH_HASH='${TWITCH_HASH}'"
```
Comment (norihiro) [inline CI/linux/02_build_obs.sh:37]:
```suggestion
        TWITCH_OPTIONS=(-DTWITCH_CLIENTID="${TWITCH_CLIENTID}" -DTWITCH_HASH="${TWITCH_HASH}")
```

File: CI/linux/02_build_obs.sh [hunk 2]
```diff
@@ -34,15 +34,15 @@ _configure_obs() {
...
     fi
 
     if [ "${RESTREAM_CLIENTID}" -a "${RESTREAM_HASH}" ]; then
-        RESTREAM_OPTIONS="-DRESTREAM_CLIENTID=\"${RESTREAM_CLIENTID}\" -DRESTREAM_HASH=\"${RESTREAM_HASH}\""
+        RESTREAM_OPTIONS="-DRESTREAM_CLIENTID='${RESTREAM_CLIENTID}' -DRESTREAM_HASH='${RESTREAM_HASH}'"
```
Comment (norihiro) [inline CI/linux/02_build_obs.sh:41]:
```suggestion
        RESTREAM_OPTIONS=(-DRESTREAM_CLIENTID="${RESTREAM_CLIENTID}" -DRESTREAM_HASH="${RESTREAM_HASH}")
```

File: CI/linux/02_build_obs.sh [hunk 3]
```diff
@@ -34,15 +34,15 @@ _configure_obs() {
...
     fi
 
     if [ "${YOUTUBE_CLIENTID}" -a "${YOUTUBE_CLIENTID_HASH}" -a "${YOUTUBE_SECRET}" -a "{YOUTUBE_SECRET_HASH}" ]; then
-        YOUTUBE_OPTIONS="-DYOUTUBE_CLIENTID=\"${YOUTUBE_CLIENTID}\" -DYOUTUBE_CLIENTID_HASH=\"${YOUTUBE_CLIENTID_HASH}\" -DYOUTUBE_SECRET=\"${YOUTUBE_SECRET}\" -DYOUTUBE_SECRET_HASH=\"${YOUTUBE_SECRET_HASH}\""
+        YOUTUBE_OPTIONS="-DYOUTUBE_CLIENTID='${YOUTUBE_CLIENTID}' -DYOUTUBE_CLIENTID_HASH='${YOUTUBE_CLIENTID_HASH}' -DYOUTUBE_SECRET='${YOUTUBE_SECRET}' -DYOUTUBE_SECRET_HASH='${YOUTUBE_SECRET_HASH}'"
```
Comment (norihiro) [inline CI/linux/02_build_obs.sh:45]:
```suggestion
        YOUTUBE_OPTIONS=(-DYOUTUBE_CLIENTID="${YOUTUBE_CLIENTID}" -DYOUTUBE_CLIENTID_HASH="${YOUTUBE_CLIENTID_HASH}" -DYOUTUBE_SECRET="${YOUTUBE_SECRET}" -DYOUTUBE_SECRET_HASH="${YOUTUBE_SECRET_HASH}")
```

File: CI/macos/02_build_obs.sh [hunk 4]
```diff
@@ -48,15 +48,15 @@ _configure_obs() {
     check_ccache
 
     if [ "${TWITCH_CLIENTID}" -a "${TWICH_HASH}" ]; then
-        TWITCH_OPTIONS="-DTWITCH_CLIENTID=\"${TWITCH_CLIENTID}\" -DTWITCH_HASH=\"${TWITCH_HASH}\""
+        TWITCH_OPTIONS="-DTWITCH_CLIENTID='${TWITCH_CLIENTID}' -DTWITCH_HASH='${TWITCH_HASH}'"
```
Comment (norihiro) [inline CI/macos/02_build_obs.sh:51]:
```suggestion
        TWITCH_OPTIONS=(-DTWITCH_CLIENTID="${TWITCH_CLIENTID}" -DTWITCH_HASH="${TWITCH_HASH}")
```

File: CI/macos/02_build_obs.sh [hunk 5]
```diff
@@ -48,15 +48,15 @@ _configure_obs() {
...
     fi
 
     if [ "${RESTREAM_CLIENTID}" -a "${RESTREAM_HASH}" ]; then
-        RESTREAM_OPTIONS="-DRESTREAM_CLIENTID=\"${RESTREAM_CLIENTID}\" -DRESTREAM_HASH=\"${RESTREAM_HASH}\""
+        RESTREAM_OPTIONS="-DRESTREAM_CLIENTID='${RESTREAM_CLIENTID}' -DRESTREAM_HASH='${RESTREAM_HASH}'"
```
Comment (norihiro) [inline CI/macos/02_build_obs.sh:55]:
```suggestion
        RESTREAM_OPTIONS=(-DRESTREAM_CLIENTID="${RESTREAM_CLIENTID}" -DRESTREAM_HASH="${RESTREAM_HASH}")
```

File: CI/macos/02_build_obs.sh [hunk 6]
```diff
@@ -48,15 +48,15 @@ _configure_obs() {
...
     fi
 
     if [ "${YOUTUBE_CLIENTID}" -a "${YOUTUBE_CLIENTID_HASH}" -a "${YOUTUBE_SECRET}" -a "{YOUTUBE_SECRET_HASH}" ]; then
-        YOUTUBE_OPTIONS="-DYOUTUBE_CLIENTID=\"${YOUTUBE_CLIENTID}\" -DYOUTUBE_CLIENTID_HASH=\"${YOUTUBE_CLIENTID_HASH}\" -DYOUTUBE_SECRET=\"${YOUTUBE_SECRET}\" -DYOUTUBE_SECRET_HASH=\"${YOUTUBE_SECRET_HASH}\""
+        YOUTUBE_OPTIONS="-DYOUTUBE_CLIENTID='${YOUTUBE_CLIENTID}' -DYOUTUBE_CLIENTID_HASH='${YOUTUBE_CLIENTID_HASH}' -DYOUTUBE_SECRET='${YOUTUBE_SECRET}' -DYOUTUBE_SECRET_HASH='${YOUTUBE_SECRET_HASH}'"
```
Comment (norihiro) [inline CI/macos/02_build_obs.sh:59]:
```suggestion
        YOUTUBE_OPTIONS=(-DYOUTUBE_CLIENTID="${YOUTUBE_CLIENTID}" -DYOUTUBE_CLIENTID_HASH="${YOUTUBE_CLIENTID_HASH}" -DYOUTUBE_SECRET="${YOUTUBE_SECRET}" -DYOUTUBE_SECRET_HASH="${YOUTUBE_SECRET_HASH}")
```

**Review by norihiro — COMMENTED** [review]
I'd like to suggest using array.
When using these array variables later, the variables should have `[@]` and double-quoted.
```patch
-        ${YOUTUBE_OPTIONS} \
-        ${TWITCH_OPTIONS} \
-        ${RESTREAM_OPTIONS} \
+        "${YOUTUBE_OPTIONS[@]}" \
+        "${TWITCH_OPTIONS[@]}" \
+        "${RESTREAM_OPTIONS[@]}" \
```

**Comment by PatTheMav** [comment]
I'd rather migrate the scripts to Zsh which has much better array handling at some later point, but for now it's enough to "unbreak" the master branch.

@norihiro Feel free to open a PR with revisions if you can get this to work. I had many issues with our Twitch and YouTube credentials breaking bash' parser in unpredictable ways.

**Pull request merged**